### PR TITLE
SAP2000_Toolkit: Fix RigidLink and misc. other bugs

### LINKED ISSUE #104: SAP2000_Toolkit: always return dummy element/property when pull fails.
#### Broken rules:
For CreateNonExisting, adapter must return an element for every element pulled. This means failure to pull a given element should trigger return of a dummy element.

### LINKED ISSUE #106: SAP2000_Toolkit: push rigidLinks fails
#### Description:
List of sublinks and list of nodes are declared as null lists, and then items are appended with `list.Add()`.  This fails. The lists should be declared as `new List<type>`

### LINKED ISSUE #110: SAP2000_Toolkit: some panels not pulling

### PR DESCRIPTION
**Pull request by JosefTaylor**
### Issues addressed by this PR

Closes #106 
Closes #104 
Closes #110 

- Fixed pull and push of RigidLinks.
- Added push LinkConstraint (previously only pull).
- Fixed pull SurfaceProperty (and thus pull panels).
- Removed refresh view API call, increasing speed of push.
- Brought surfaceProperty modifiers into compliance.

### Test files
https://burohappold.sharepoint.com/sites/BHoM/_layouts/15/SkySyncRedir.aspx?Type=2&ResourceId=d033310bceb74eadb6675e145e7ce6ca

### CODE REVIEW

File: SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs [hunk 1]
```diff
@@ -61,42 +61,35 @@ private List<ISurfaceProperty> ReadSurfaceProperty(List<string> ids = null)
...
+                panelConstant.CustomData.Add("MembraneF11Modifier", modifiers[0]);
+                panelConstant.CustomData.Add("MembraneF22Modifier", modifiers[1]);
+                panelConstant.CustomData.Add("MembraneF12Modifier", modifiers[2]);
+                panelConstant.CustomData.Add("BendingM11Modifier", modifiers[3]);
+                panelConstant.CustomData.Add("BendingM22Modifier", modifiers[4]);
+                panelConstant.CustomData.Add("BendingM12Modifier", modifiers[5]);
+                panelConstant.CustomData.Add("ShearV13Modifier", modifiers[6]);
+                panelConstant.CustomData.Add("ShearV23Modifier", modifiers[7]);
+                panelConstant.CustomData.Add("MassModifier", modifiers[8]);
+                panelConstant.CustomData.Add("WeightModifier", modifiers[9]);
```
Comment (IsakNaslundBh) [inline SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs:90]: We have a method called "ApplyModifiers" in Structure_Engine that is meant to be used for this.

Will formalise this into a fragment this milestone, is the plan at least.

File: SAP2000_Engine/Query/SplitRigidLink.cs [hunk 2]
```diff
@@ -36,24 +36,24 @@ public static partial class Query
 
         public static List<RigidLink> SplitRigidLink(RigidLink link)
         {
-            List<RigidLink> links = null;
+            List<RigidLink> SingleLinks = new List<RigidLink>();
```
Comment (IsakNaslundBh) [inline SAP2000_Engine/Query/SplitRigidLink.cs:39]: minor comment, but should be lowercase letter

File: SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs (hunk 1 above)
Comment (JosefTaylor) [reply SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs:90]: Ah, thanks! can I use it now or should I wait till fragment is released?

File: SAP2000_Engine/Query/SplitRigidLink.cs (hunk 2 above)
Comment (JosefTaylor) [reply SAP2000_Engine/Query/SplitRigidLink.cs:39]: Yep, you're right!

File: SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs (hunk 1 above)
Comment (JosefTaylor) [reply SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs]: I can use it now! Thanks @IsakNaslundBh

Comment (IsakNaslundBh) [reply SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs]: answered yourself, but yet. Was planning to make those methods point to the fragment as soon as it has been added :)

**Comment by JosefTaylor** [comment]
@FraserGreenroyd I've fixed the references and Compliance runs fine as you can see. Does that hit everything you noted? SAP2000v1 was meant to be a NuGet reference but it looks like I hadn't got it quite right. Now it should be OK.

**Comment by JosefTaylor** [comment]
@enarhi mind taking a final review?

**Pull request merged**